py/server/deephaven/stream/kafka/consumer.py

```diff
@@ -161,6 +163,65 @@ def consume(
         DHError
     """
 
+    return _consume(kafka_config, topic, partitions, offsets, key_spec, value_spec, table_type, to_partitioned=False)
+
+
+def consume_to_partitioned_table(
+        kafka_config: Dict,
+        topic: str,
+        partitions: List[int] = None,
+        offsets: Dict[int, int] = None,
+        key_spec: KeyValueSpec = None,
+        value_spec: KeyValueSpec = None,
+        table_type: TableType = TableType.stream(),
+) -> PartitionedTable:
+    """Consume from Kafka to a Deephaven partitioned table.
+
+    Args:
+        kafka_config (Dict): configuration for the associated Kafka consumer and also the resulting table.
+            Once the table-specific properties are stripped, the remaining one is used to call the constructor of
+            org.apache.kafka.clients.consumer.KafkaConsumer; pass any KafkaConsumer specific desired configuration here
+        topic (str): the Kafka topic name
+        partitions (List[int]) : a list of integer partition numbers, default is None which means all partitions
+        offsets (Dict[int, int]) : a mapping between partition numbers and offset numbers, and can be one of the
+            predefined ALL_PARTITIONS_SEEK_TO_BEGINNING, ALL_PARTITIONS_SEEK_TO_END or ALL_PARTITIONS_DONT_SEEK.
+            The default is None which works the same as  ALL_PARTITIONS_DONT_SEEK. The offset numbers may be one
+            of the predefined SEEK_TO_BEGINNING, SEEK_TO_END, or DONT_SEEK.
+        key_spec (KeyValueSpec): specifies how to map the Key field in Kafka messages to Deephaven column(s).
+            It can be the result of calling one of the functions: simple_spec(),avro_spec() or json_spec() in this
+            module, or the predefined KeyValueSpec.IGNORE or KeyValueSpec.FROM_PROPERTIES. The default is None which
+            works the same as KeyValueSpec.FROM_PROPERTIES, in which case, the kafka_config param should include values
+            for dictionary keys 'deephaven.key.column.name' and 'deephaven.key.column.type', for the single resulting
+            column name and type
+        value_spec (KeyValueSpec): specifies how to map the Value field in Kafka messages to Deephaven column(s).
+            It can be the result of calling one of the functions: simple_spec(),avro_spec() or json_spec() in this
+            module, or the predefined KeyValueSpec.IGNORE or KeyValueSpec.FROM_PROPERTIES. The default is None which
+            works the same as KeyValueSpec.FROM_PROPERTIES, in which case, the kafka_config param should include values
+            for dictionary keys 'deephaven.key.column.name' and 'deephaven.key.column.type', for the single resulting
+            column name and type
+        table_type (TableType): a TableType enum, specifying the type of the expected result's constituent tables,
+            default is TableType.stream()
+
+    Returns:
+        a Deephaven live partitioned table that will update based on Kafka messages consumed for the given topic
```

Comment:
I think we should clarify that the keys are ints from the topic partition, and the constituents are tables per topic partition.